### PR TITLE
Microstack installation instructions on openstack page

### PR DESCRIPTION
**Pull request by carkod**
## Done

Get openstack install instructions from microstack.run/docs

## QA

- Check https://ubuntu-com-10061.demos.haus/openstack/install compare with current https://ubuntu.com/openstack/install
Styling is done by discourse module, so it cannot be modified. Docs content comes from https://microstack.run/docs

- [Tabs UX wireframe](https://miro.com/app/board/o9J_l5Llu-A=/)

## Issue / Card

Fixes https://github.com/canonical-web-and-design/web-squad/issues/4263

### CODE REVIEW

File: webapp/views.py [hunk 1]
```diff
@@ -637,6 +642,51 @@ def render_template(language, page):
...
+
+    singlenode_topic = openstack_install_docs.parser.api.get_topic(21427)
+    singlenode_topic_soup = BeautifulSoup(
+        singlenode_topic["post_stream"]["posts"][0]["cooked"],
+        features="html.parser",
+    )
+    singlenode_content = openstack_install_docs.parser._process_topic_soup(
+        singlenode_topic_soup
+    )
+
```
Comment (carkod) [inline webapp/views.py:673]: Fix lightbox issue

**Comment by davegoddard42** [comment]
We don't need the links at the bottom of the grey section any more @carkod. Otherwise fine

image

**Comment by davegoddard42** [comment]
@carkod - looks great, thanks

File: templates/openstack/install.html [hunk 2]
```diff
@@ -39,54 +38,82 @@ <h1>Install OpenStack yourself</h1>
 </section>
 
 <section class="p-strip--light is-bordered">
-  <div class="row">
-    <div class="col-8">
-      <h2>Single-node, multi-node or large scale?</h2>
-      <p><strong>Choose the OpenStack installation option that suits you best:</strong></p>
+  <div class="row u-equal-height">
```
Comment (petesfrench) [inline templates/openstack/install.html:41]: Only col's should be direct children of rows, you can change this to a u-fixed-width or nest a col-12 within with everything else in that.

File: templates/openstack/install.html [hunk 3]
```diff
@@ -39,54 +38,82 @@ <h1>Install OpenStack yourself</h1>
...
+        <div class="p-tabs__item">
+          <button style="background-color: transparent;" class="p-tabs__link" role="tab" aria-selected="false" aria-controls="multi-node-deployment" id="multi-node-deployment-tab" tabindex="-1">Multi&dash;node deployment</button>
+        </div>
+        <div class="p-tabs__item">
+          <button style="background-color: transparent;" class="p-tabs__link" role="tab" aria-selected="false" aria-controls="large-scale-deployment" id="large-scale-deployment-tab" tabindex="-1">Large-scale deployment</button>
+        </div>
+      </div>
     </div>
   </div>
   <div class="row u-equal-height">
```
Comment (petesfrench) [inline templates/openstack/install.html:60]: Same here, u-fixed-width

File: templates/openstack/install.html [hunk 4]
```diff
@@ -115,24 +143,27 @@ <h2>Multi-node OpenStack deployment</h2>
...
-    </p>
-    <p class="u-sv3"><a href="https://docs.openstack.org/project-deploy-guide/charm-deployment-guide/latest/" class="p-link--external">OpenStack Charms Deployment Guide here</a>.</p>
-    <div class="p-notification">
-      <p class="p-notification__response">
-        <span class="p-notification__status">NOTE:</span> In order to qualify for the <a href="https://assets.ubuntu.com/v1/d4766546-Private-Cloud-Build_Datasheet.pdf" class="p-link--external">Private Cloud Build (PCB)</a> service, <a href="https://assets.ubuntu.com/v1/1a8fb1b3-UA-I_datasheet_2019-Oct.pdf" class="p-link--external">Ubuntu Advantage for Infrastructure (UA-I)</a> support subscription and <a href="https://assets.ubuntu.com/v1/a797f6f9-Datasheet_Openstack_05.10.20+%281%29.pdf" class="p-link--external">Managed OpenStack</a>, at least 12 nodes are required for Charmed OpenStack. Please contact your Canonical sales representative for detailed information on minimum hardware requirements.
+<section class="p-strip is-bordered" id="large-scale-deployment" hidden="true">
+  <div class="row u-equal-heigh">
```
Comment (petesfrench) [inline templates/openstack/install.html:153]: Same here, u-fixed-width

**Review by petesfrench — CHANGES_REQUESTED** [review]
Just a couple semantic changes to fit with vanilla patterns. 

Also, I couldn't use the tab key to change between the 'single-node', 'multi-node' and 'large-scale'.

File: templates/openstack/install.html (hunk 2 above)
Comment (carkod) [reply templates/openstack/install.html]: Thanks, I was actually looking for this class. Got it wrong.

**Comment by carkod** [comment]
> Just a couple semantic changes to fit with vanilla patterns.
> 
> Also, I couldn't use the tab key to change between the 'single-node', 'multi-node' and 'large-scale'.

So I don't think the tab key is supported at this point in the [Vanilla tab pattern](https://vanillaframework.io/docs/patterns/tabs#content). If you believe this is required, you should [open an issue on Vanilla](https://github.com/canonical-web-and-design/vanilla-framework/issues), as this is something Vanilla would include.

Also my implementation is a bit different, because the tabs trigger a different `<section>` due to how the [design](https://miro.com/app/board/o9J_l5Llu-A=/) separates the content (the Vanilla pattern requires tab content to be immediately after the tab list, but because the design uses a separate strip, I had to tweak the default JS functionality)

File: templates/openstack/install.html [hunk 5]
```diff
@@ -39,105 +38,137 @@ <h1>Install OpenStack yourself</h1>
...
 <section class="p-strip--light is-bordered">
-  <div class="row">
-    <div class="col-8">
-      <h2>Single-node, multi-node or large scale?</h2>
-      <p><strong>Choose the OpenStack installation option that suits you best:</strong></p>
+  <div class="u-fixed-width">
+    <h2>Single-node, multi-node or large-scale cluster?</h2>
+    <p><strong>Choose the OpenStack installation option that suits you best:</strong></p>
+  </div>
+  <div class="p-tabs">
```
Comment (petesfrench) [inline templates/openstack/install.html:45]: The u-fixed-width should be wrapping the p-tabs div.

File: templates/openstack/install.html [hunk 6]
```diff
@@ -39,105 +38,137 @@ <h1>Install OpenStack yourself</h1>
...
-        <li class="p-list__item is-ticked">Core services included</li>
-        <li class="p-list__item is-ticked">All services on a single node</li>
-        <li class="p-list__item is-ticked">Opinionated OpenStack</li>
-        <li class="p-list__item is-ticked">Straightforward installation</li>
-        <li class="p-list__item is-ticked">OpenStack "on-rails"</li>
-      </ul>
-      <p><a href="#single-node-deployment">Single-node installation instructions &rsaquo; Deploy OpenStack on a single node&nbsp;&rsaquo;</a></p>
+  <div class="u-fixed-width">
+    <div tabindex="0" role="tabpanel" id="single-node-deployment" aria-labelledby="single-node-deployment">
+      <div class="row">
```
Comment (petesfrench) [inline templates/openstack/install.html:62]: If you change the div on line 60 to a row, you dont need the row div on line 62.

**Review by petesfrench — APPROVED** [review]
LGTM, just one little change :+1:

**Comment by sowasred2012** [comment]
> So I don't think the tab key is supported at this point in the [Vanilla tab pattern](https://vanillaframework.io/docs/patterns/tabs#content). If you believe this is required, you should [open an issue on Vanilla](https://github.com/canonical-web-and-design/vanilla-framework/issues), as this is something Vanilla would include.
> 
> Also my implementation is a bit different, because the tabs trigger a different `<section>` due to how the [design](https://miro.com/app/board/o9J_l5Llu-A=/) separates the content (the Vanilla pattern requires tab content to be immediately after the tab list, but because the design uses a separate strip, I had to tweak the default JS functionality)

@carkod You're right that the tab key isn't supported, but it should be possible to navigate between these tabs using arrow keys, otherwise there's no way for keyboard only users to access the content in the other tabs: https://vanillaframework.io/docs/patterns/tabs#content

Those docs link to this example on the w3, regarding how to make tabbed content accessible: https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-1/tabs.html

I will add an issue to Vanilla to make that more explicit in our docs.

**Comment by carkod** [comment]
> > So I don't think the tab key is supported at this point in the [Vanilla tab pattern](https://vanillaframework.io/docs/patterns/tabs#content). If you believe this is required, you should [open an issue on Vanilla](https://github.com/canonical-web-and-design/vanilla-framework/issues), as this is something Vanilla would include.
> > Also my implementation is a bit different, because the tabs trigger a different `<section>` due to how the [design](https://miro.com/app/board/o9J_l5Llu-A=/) separates the content (the Vanilla pattern requires tab content to be immediately after the tab list, but because the design uses a separate strip, I had to tweak the default JS functionality)
> 
> @carkod You're right that the tab key isn't supported, but it should be possible to navigate between these tabs using arrow keys, otherwise there's no way for keyboard only users to access the content in the other tabs: https://vanillaframework.io/docs/patterns/tabs#content - that code snippet provides example JS for listening for arrow key events.
> 
> Those docs link to this example on the w3, regarding how to make tabbed content accessible: https://www.w3.org/TR/wai-aria-practices-1.1/examples/tabs/tabs-1/tabs.html
> 
> I will add an issue to Vanilla to make that more explicit in our docs.

Thanks for raising that. I added support for arrow keys.

**Review by sowasred2012 — APPROVED** [review]
Thanks for that @carkod - LGTM!

**Pull request merged**